h2o-core/src/main/java/water/H2O.java

```diff
@@ -591,7 +591,7 @@ else if (s.matches("nthreads")) {
       }
       else if (s.matches("hdfs_config")) {
         i = s.incrementAndCheck(i, args);
-        trgt.hdfs_config = args[i];
+        trgt.hdfs_config = ArrayUtils.append(trgt.hdfs_config, args[i]);
```

Comment:
Enlarging the array here is fine, as we don't know the number of HDFS confings in advance.